include/sys/arch_inlines.h

```diff
@@ -0,0 +1,460 @@
+ */
+
+/**
+ * @def Z_ARCH_THREAD_STACK_RESERVED
+ *
+ * @see K_THREAD_STACK_RESERVED
+ */
+
+/**
+ * @def Z_ARCH_THREAD_STACK_BUFFER(sym)
+ *
+ * @see K_THREAD_STACK_RESERVED
+ */
+
+/** @} */
+
+
+/**
+ * @addtogroup arch-pm
+ * @{
+ */
+
+/**
+ * @brief Power save idle routine
+ *
+ * This function will be called by the kernel idle loop or possibly within
+ * an implementation of z_sys_power_save_idle in the kernel when the
+ * '_sys_power_save_flag' variable is non-zero.
+ *
+ * Architectures that do not implement power management instructions may
+ * immediately return, otherwise a power-saving instruction should be
+ * issued to wait for an interrupt.
+ *
+ * @see k_cpu_idle()
+ */
+void z_arch_cpu_idle(void);
+
+/**
+ * @brief Atomically re-enable interrupts and enter low power mode
+ *
+ * The requirements for z_arch_cpu_atomic_idle() are as follows:
+ * 1) The enablement of interrupts and entering a low-power mode needs to be
+ *    atomic, i.e. there should be no period of time where interrupts are
+ *    enabled before the processor enters a low-power mode.  See the comments
+ *    in k_lifo_get(), for example, of the race condition that
+ *    occurs if this requirement is not met.
+ *
+ * 2) After waking up from the low-power mode, the interrupt lockout state
+ *    must be restored as indicated in the 'key' input parameter.
+ *
+ * @see k_cpu_atomic_idle()
+ *
+ * @param key Lockout key returned by previous invocation of z_arch_irq_lock()
+ */
+void z_arch_cpu_atomic_idle(unsigned int key);
+
+/** @} */
+
+
+/**
+ * @addtogroup arch-smp
+ * @{
+ */
+
+/**
+ * @brief Start a numbered CPU on a MP-capable system
+ *
+ * This starts and initializes a specific CPU.  The main thread on startup is
+ * running on CPU zero, other processors are numbered sequentially.  On return
+ * from this function, the CPU is known to have begun operating and will enter
+ * the provided function.  Its interrupts will be initialized but disabled such
+ * that irq_unlock() with the provided key will work to enable them.
+ *
+ * Normally, in SMP mode this function will be called by the kernel
+ * initialization and should not be used as a user API.  But it is defined here
+ * for special-purpose apps which want Zephyr running on one core and to use
+ * others for design-specific processing.
+ *
+ * @param cpu_num Integer number of the CPU
+ * @param stack Stack memory for the CPU
+ * @param sz Stack buffer size, in bytes
+ * @param fn Function to begin running on the CPU.  First argument is
+ *        an irq_unlock() key.
+ * @param arg Untyped argument to be passed to "fn"
+ */
+void z_arch_start_cpu(int cpu_num, k_thread_stack_t *stack, int sz,
+		      void (*fn)(int key, void *data), void *arg);
+/** @} */
+
+
+/**
+ * @addtogroup arch-irq
+ * @{
+ */
+
+/**
+ * Lock interrupts on the current CPU
+ *
+ * @see irq_lock()
+ */
+static inline unsigned int z_arch_irq_lock(void);
+
+/**
+ * Unlock interrupts on the current CPU
+ *
+ * @see irq_unlock()
```

Comment:
When you hook this up to the API documentation (via Sphinx directives), I suspect doxygen won't be happy about the lack of an ``@param key`` description, and throw a "missing parameter documentation" error.

Reply:
Is there a way to get these to inherit the parameter definition from another doxgen? I'd like this to inherit the parameter list from irq_unlock() (and everywhere else I am using @see)

Reply:
BTW doxygen is not raising warnings about this AFAICT, I have another branch where it does scan all this stuff.